sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEPropagateEmptyRelation.scala

```diff
@@ -40,6 +41,16 @@ object AQEPropagateEmptyRelation extends PropagateEmptyRelationBase {
   private def getRowCount(plan: LogicalPlan): Option[BigInt] = plan match {
```

Comment:
I'd suggest renaming this to `getEstimatedRowCount` with a comment to say that it can be over estimated if row count is not 0, to match the new change.